src/gui/properties/peerlistwidget.cpp

```diff
@@ -138,6 +136,67 @@ PeerListWidget::~PeerListWidget()
     delete m_copyHotkey;
 }
 
+void PeerListWidget::displayToggleColumnsMenu(const QPoint&)
+{
+    QMenu hideshowColumn(this);
+    hideshowColumn.setTitle(tr("Column visibility"));
+    QList<QAction*> actions;
+    for (int i = 0; i < PeerListDelegate::IP_HIDDEN; ++i) {
+        QAction *myAct = hideshowColumn.addAction(this->headerData(i, Qt::Horizontal, Qt::DisplayRole).toString());
+        myAct->setCheckable(true);
+        myAct->setChecked(!isColumnHidden(i));
+        actions.append(myAct);
+    }
+    int visibleCols = 0;
+    for (unsigned int i = 0; i < PeerListDelegate::COL_COUNT; i++) {
+        if (!isColumnHidden(i))
+            visibleCols++;
+
+        if (visibleCols > 1)
+            break;
+    }
+
+    // Call menu
+    QAction *act = hideshowColumn.exec(QCursor::pos());
+    if (act) {
+        int col = actions.indexOf(act);
+        Q_ASSERT(col >= 0);
+        Q_ASSERT(visibleCols > 0);
+        if (!isColumnHidden(col) && visibleCols == 1)
+            return;
+        qDebug("Toggling column %d visibility", col);
+        setColumnHidden(col, !isColumnHidden(col));
+        if (!isColumnHidden(col) && columnWidth(col) <= 5)
+            setColumnWidth(col, 100);
+        saveSettings();
+    }
+}
+
+QVariant PeerListWidget::headerData(int section, Qt::Orientation orientation, int role) const
+{
+    if (orientation == Qt::Horizontal) {
+        if (role == Qt::DisplayRole) {
```

Comment:
Instead to put almost all the function code at the 3rd nesting level, it might be easier in this case:
```c++
if ((orientation != Qt::Horizontal) || (role != Qt::DisplayRole)) return QVariant();

// the rest code
```